documentation/adr/0008-pace-syntax-guiding-principles.md

```diff
@@ -0,0 +1,44 @@
+[<-previous](0007-developer-scripts-storage-location.md) | next->
+
+# 8. PACE Syntax Guiding Principles
+
+Date: 2020-Oct-19
+
+## Status
+
+Proposed
+
+
+## Context
+
+Whilst the core framework of PACE is the Horace (and Herbert) projects,
+PACE involves many other codes, including [Euphonic](https://github.com/pace-neutrons/Euphonic),
+[Brille](https://github.com/brille/brille) and [SpinW](https://spinw.org).
+The way in which these programs interact with each other is presently unclear.
+Furthermore the programs are written in several different languages.
+In this ADR we propose some guiding principles in the design of the interfaces
+between the programs or components which make up PACE.
+
+## Decision
+
+The guiding principle for PACE-supported programs is **commonality**.
+That is, interacting with the different programs of PACE should be **seamless**
+to the user, and the user interface presented to them should be **uniform**.
+
+## Consequences
+
+There are several implications stemming from this guiding principle:
+
+* There should be a single downloadble distribution which bundles all the 
+necessary components (e.g. Horace, Euphonic, Brille and SpinW at the moment).
```

Comment:
Alternatively, replace `e.g.` by `i.e.,` which is 'that is.'

On the point of a single distribution: do we want to commit to packaging disparate Python and MATLAB projects into a single archive (e.g., `.tar.gz`, `.zip`, `.exe`, or the like) or would an installation _script_ be a better alternative?
At the moment `Euphonic` and `Brille` are easily installed using `pip` and I wonder if the something similar could be accomplished for `Horace/Herbert` and `spinW` through MATLAB Toolboxes.

Reply:
```suggestion
necessary components (currently: Horace, Euphonic, Brille and SpinW).
```

Reply:
I think we would have to have separate Python and Matlab packages.

The problem is bundling everything together:
* For the Matlab distribution, how do we install the Python packages?
  - It basically depends on what distribution of Python the user wants to use. The `pyenv` and `pyversion` commands can be set up to use any install Python. The easiest might be to have a script to ask the user which Pythons they want and then to install the Python dependencies using `pip` to that Python distribution. This could accommodate people wanting to use Mantid's Python, Anaconda or the system Python.
  - The alternative is that we bundle our own Python distribution in an executable installer (for Windows) or create a package dependency on the system Python and then use that (Linux packagers).
* For the Python distribution, how to install the Matlab component?
  - This is a bit more straightforward since it definitely requires the Matlab MCR. The choice here is between bundling it or asking users to download it themselves. Bundling it means our build artifacts will be bigger by ~2GB. And I'm not so sure about the licensing for this. If you use their deployment tool, Matlab allows you to bundle the MCR but I'm not sure they allow you to do it in your own installers...
   - The alternative is to tell users they have to install the MCR themselves.
   - The third way is to have the installation script download the MCR for users and start running it so they just have to click-through the license.

I sense another meeting about this...